Backend: Create SignUtils

rueblimaster opened the pull request:
## What
Move sign related utils into newly created SignUtils


## Changelog Technical Details
+ Created SignUtils. - rueblimaster

Commit:
Add Optimal Angles Overlay for Squeaky Mousemat (like Rancher Boots Overlay)

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
fix events merge

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

nikijul committed:
Add SignUtils and relocate suitable affected functions

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/nikijul007/feature_garden_angles

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/utils/LorenzUtils.kt

CalMWolfs committed:
fix merge

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Rename .java to .kt

CalMWolfs committed:
make kotlin, dictionary

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/nikijul007/feature_garden_angles

CalMWolfs committed:
detekt

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

CalMWolfs committed:
Merge branch 'beta' into feature_garden_angles

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/nikijul007/feature_garden_angles

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/data/GuiEditManager.kt
#	src/main/java/at/hannibal2/skyhanni/features/garden/GardenOptimalSpeed.kt
#	src/main/java/at/hannibal2/skyhanni/utils/LorenzUtils.kt

hannibal002 committed:
Fixed merge conflicts

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/nikijul007/feature_garden_angles

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/features/misc/BetterSignEditing.kt

rueblimaster committed:
test

rueblimaster committed:
revert

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
revert step 1

rueblimaster committed:
revert step 2

rueblimaster committed:
revert step 3

rueblimaster committed:
revert step 4

rueblimaster committed:
revert step 5

rueblimaster committed:
wierd formatting?

rueblimaster committed:
detekt

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

Thunderblade73 commented on src/main/java/at/hannibal2/skyhanni/utils/LorenzUtils.kt:
```diff
@@ -224,11 +204,32 @@ object LorenzUtils {
-
-        val signText = (this as AccessorGuiEditSign).tileSign.signText.map { it.unformattedText.removeColor() }
-        return signText[1] == "^^^^^^" && signText[2] == "Set your" && signText[3] == "speed cap!"
+    @Deprecated("do not use List<Any>, use List<Renderable> instead", ReplaceWith(""))
+    inline fun MutableList<List<Any>>.addButton(
+        prefix: String,
+        getName: String,
+        crossinline onChange: () -> Unit,
+        tips: List<String> = emptyList(),
+    ) {
+        val onClick = {
+            if ((System.currentTimeMillis() - ChatUtils.lastButtonClicked) > 150) { // funny thing happen if I don't do that
+                onChange()
+                SoundUtils.playClickSound()
+                ChatUtils.lastButtonClicked = System.currentTimeMillis()
+            }
+        }
+        add(
+            buildList {
+                add(prefix)
+                add("§a[")
+                if (tips.isEmpty()) {
+                    add(Renderable.link("§e$getName", false, onClick))
+                } else {
+                    add(Renderable.clickAndHover("§e$getName", tips, false, onClick))
+                }
+                add("§a]")
+            },
+        )
```
This is probably unintentional. Since that looks like a bad merge conflict handle.

rueblimaster committed:
Remove bad merge

rueblimaster commented:
@Thunderblade73 I re - request a review

Thunderblade73 commented:
> @Thunderblade73 I re - request a review

![image](https://github.com/user-attachments/assets/d3a07bc2-e634-456e-8630-f87b6f00d396)
This button is enough. You don't have to ping.

hannibal002 commented:
omg no dark mode

rueblimaster commented:
> omg no dark mode

Oh shit, you're right

rueblimaster commented:
> > @Thunderblade73 I re - request a review
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/d3a07bc2-e634-456e-8630-f87b6f00d396)
> 
> This button is enough. You don't have to ping.

Haven't found the button on mobile yet...

Thunderblade73 commented:
> > > @Thunderblade73 I re - request a review
> > 
> > 
> > ![image](https://private-user-images.githubusercontent.com/85900443/411320067-d3a07bc2-e634-456e-8630-f87b6f00d396.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkxMTg5NzEsIm5iZiI6MTczOTExODY3MSwicGF0aCI6Ii84NTkwMDQ0My80MTEzMjAwNjctZDNhMDdiYzItZTYzNC00NTZlLTg2MzAtZjg3YjZmMDBkMzk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA5VDE2MzExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlMDlhZDY4OWRlZmNhNzg3NDM5NGI1NWY5MGQwZjhmNGI5Y2ZlN2QzNTc5MGU1YmNlZTU1NmRiOWY0MzZhMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WIHkT0x2JFSGm-x_nd0ltFbyolrg5Zmp98_TYRQXbeA)
> > This button is enough. You don't have to ping.
> 
> Haven't found the button on mobile yet...

At the bottom:
![image](https://github.com/user-attachments/assets/73d9987d-4313-44c1-91cd-4505e5074402)

Thunderblade73 commented on .idea/dictionaries/default_user.xml:
```diff
@@ -292,4 +292,4 @@
   </dictionary>
-</component>
\ No newline at end of file
+</component>
```
No clue what happend here. But that change should not be there.

rueblimaster replied:
you're right, this happens when you edit the file inside of github, it is now no longer a change, as it is like that normally now

Thunderblade73 reviewed (changes requested):
Looked at each line and the match up. So in total this is just a move + the isMousematSign function being added (which I can't verify that it is correct, since I don't have access to a mousemat).
Only one small random change that does not need to be there.

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
code cleanup

https://github.com/hannibal002/SkyHanni/commit/f0b0dda4cb0cdfaf256df6b48e568cf292cf98b9

rueblimaster committed:
Merge branch 'sign-utils' of https://github.com/rueblimaster/SkyHanni into sign-utils

rueblimaster commented:
so, what is this waiting for?

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster commented:
originates from: https://github.com/hannibal002/SkyHanni/pull/2999

rueblimaster committed:
Merge branch 'beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
fix stuff + detekt

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils

rueblimaster committed:
Merge branch 'hannibal002:beta' into sign-utils
